hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/snapshot/OMSnapshotCreateRequest.java

```diff
@@ -87,7 +89,15 @@ public OMSnapshotCreateRequest(OMRequest omRequest) {
   }
 
   @Override
+  @DisallowedUntilLayoutVersion(SNAPSHOT_SUPPORT)
   public OMRequest preExecute(OzoneManager ozoneManager) throws IOException {
+    if (!ozoneManager.getVersionManager()
+            .isAllowed(SNAPSHOT_SUPPORT)) {
+      throw new OMException(
+              "cannot be invoked before finalization.",
```

Comment:
The annotation `@DisallowedUntilLayoutVersion` itself is already supposed to do the same thing as the `isAllowed()` check internally. It will throw exception if the layout version is too low.

So this `if` condition is unnecessary here.

Reply:
Got it! sure, let me remove the if-else condition!

Reply:
fyi this is the code that does the check (AspectJ): https://github.com/apache/ozone/blob/92daac20f0375ecaff2f6b5aa5073c5f99f13d89/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/upgrade/OMLayoutFeatureAspect.java#L53-L79